go/vt/vtgate/planbuilder/operators/query_planning.go

```diff
@@ -107,6 +107,13 @@ func runRewriters(ctx *plancontext.PlanningContext, root Operator) Operator {
 		}
 	}
 
+	if pbm, ok := root.(*PercentBasedMirror); ok {
+		root = pbm.Clone([]Operator{
+			runRewriters(ctx, pbm.Operator),
+			runRewriters(ctx.UseMirror(), pbm.Target),
```

Comment:
We should return after this, right? We don't want to run the fixpointbottomup again right?

Reply:
if we don't, then we get this error `VT13001: [BUG] should not see *operators.Horizon here`. running fixpointbottomup takes care of getting horizons into the right place.